### PR TITLE
[pre-scan] Rename process_resources() and use it.

### LINKED ISSUE #689: How to ignore binary files ?
I've been thinking about how to speed up scans of packages, which contain big binary files.
I [can](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/workers/license.py#L20) pass patterns to `--ignore` if I know the extensions I want to ignore, but that's tedious, because there's so many of them.
For example [pypi/alfanous](https://pypi.python.org/pypi/alfanous) contains some index with huge binary files with extensions like `dcz`, `pst`, `tiz`, etc.

Also when scancode reaches timeout while scanning such file it returns non-zero exit code, which is not what I want.

So I've been thinking about something like `--ignore-binaries`, which would tell scancode to scan only text files. But I'm not sure if we can reliable detect binary/text files with Python, see.
https://stackoverflow.com/questions/898669/how-can-i-detect-if-a-file-is-binary-non-text-in-python

Ideas ?

### PR DESCRIPTION
**Pull request by jpopelka**
I've been looking at how the pre-scan plugins work in order to fix #689.

I've realized that the [process_resources()](https://github.com/nexB/scancode-toolkit/blob/develop/src/plugincode/pre_scan.py#L52) is actually never used,
so I'm adding a code which runs it.

The pre-scan plugin which does what I need would then look like (it's not part of this PR):

```python
@pre_scan_impl
class ProcessIgnoreBinaries(PreScanPlugin):
    """
    Ignore binaries, medias and files bigger than specified size.
    """
    option_attrs = dict(metavar='<size>', default=1000000)

    def __init__(self, user_input):
        super(ProcessIgnoreBinaries, self).__init__(user_input)

    def process_resource(self, resource):
        if any((resource.infos['is_binary'],
                resource.infos['is_media'],
                resource.infos['size'] > self.user_input)):
            return None
        return resource
```

### CODE REVIEW

**Comment by pombredanne** [comment]
@jpopelka good catch! :blush:  let me review and possibly add a test!

File: src/scancode/cli.py [hunk 1]
```diff
@@ -536,7 +536,7 @@ def scan(input_path,
          scans_cache_class=None,
          strip_root=False,
          full_root=False,
-         pre_scan_plugins=()):
+         pre_scan_plugins=None):
```
Comment (yashdsaraf) [inline src/scancode/cli.py]: Giving an empty tuple as a default parameter gives the reader an idea of what type of data to expect. Why did you change it?

File: src/scancode/cli.py [hunk 2]
```diff
@@ -794,7 +794,7 @@ def build_ignorer(ignores, unignores):
     return partial(ignore.is_ignored, ignores=ignores, unignores=unignores)
 
 
-def resource_paths(base_path, diag, scans_cache_class, pre_scan_plugins=()):
+def resource_paths(base_path, diag, scans_cache_class, pre_scan_plugins=None):
```
Comment (yashdsaraf) [inline src/scancode/cli.py]: See my previous comment

File: src/scancode/cli.py (hunk 1 above)
Comment (jpopelka) [reply src/scancode/cli.py]: Well, it gives a wrong idea to me since the `pre_scan_plugins` is actually a list.
I should have explain it in the commit message, sorry.

Comment (yashdsaraf) [reply src/scancode/cli.py]: It's not a good idea to pass mutable objects as default parameters. Using a tuple gives an idea that a collection is passed here. :)
BTW I didn't know this myself till @pombredanne posted a similar comment in an earlier review.
I'll try to find that for you. :P

Comment (jpopelka) [reply src/scancode/cli.py]: I know that. That's why I changed it to `None` and not `[]`.

Comment (pombredanne) [reply src/scancode/cli.py]: I might prefer using an empty tuple as an hint that the expected value is an iterable ofsorts, but that's a weak indication for sure... @jpopelka your way works too.

**Comment by pombredanne** [comment]
I am merging this as is... we can add tests later!
@jpopelka Thanks ++

File: src/scancode/cli.py (hunk 1 above)
Comment (pombredanne) [reply src/scancode/cli.py]: @jpopelka to your point, confusing is not a good thing, hence why I went ahead and merged it as-is :) Thanks again!
